### DISCUSSION TITLE
Snap store pointing to localhost, so not downloading packages

### DISCUSSION THREAD

**Discussion by anonimo82** [nathaneltitane/dextop#24] [Q&A] [answered]
Hello everyone.
I'm receiving this error from snap:

`error: cannot communicate with server: Post "http://localhost v2/snaps/firefox": dial unix /run/snapd.socket: connect: no such file or directory`

When trying to install any package.

Note I'm trying as user container (-u termux)

Why?

**Comment by nathaneltitane** [comment]
> Hello everyone.
> I'm receiving this error from snap:
> 
> `error: cannot communicate with server: Post "http://localhost v2/snaps/firefox": dial unix /run/snapd.socket: connect: no such file or directory`
> 
> When trying to install any package.
> 
> Note I'm trying as user container (-u termux)
> 
> Why?

Once again, look at the Readme: Snap packages are unsupported since they require the snapd service to run in background.

![image](https://user-images.githubusercontent.com/5674757/236630612-4426b62f-769e-4508-8e6f-17325bfbcd44.png)

Please use the apt/apt-get installer instead.

**Comment by anonimo82** [comment, marked as answer]
Long story short, I need Snap to use Eclipse, but since I can't have it, I switched to VHEditor ( https://github.com/vhqtvn/VHEditor-Android ), with AWebServer App ( https://play.google.com/store/apps/details?id=com.sylkat.apache ).
Now, it's not really as comfortable as Eclipse for me, but I'll live with that.

Long story:
Eclipse requires a Web Browser to be able to launch a project you're working to from its IDE, and the Browser itself you can install only with Snap (I tried `sudo apt install firefox` or `sudo apt install chromium-browser`, but both it only downloads a dummy package wich asks to download the Snap for the equivalent). Hence, the requirement for Snap.

Meanwhile, VHEditor is a port of CodeServer to Android (it's rather more of an adaptation with a WebView), and it works as a native Android App, so I can access my internal memory without weird mechanisms like termux-setup-storage.
The AWebServer App provides PHP, MySQL and Apache, enabling me to select my working folder case-by-case (based on my project folder) and allows me to code simple, yet complete Web Apps with PHP/MySQL.

Thanks anyhow, and I appreciate the time you took for helping me.

**Reply by nathaneltitane** [reply]
> Long story short, I need Snap to use Eclipse, but since I can't have it, I switched to VHEditor ( https://github.com/vhqtvn/VHEditor-Android ), with AWebServer App ( https://play.google.com/store/apps/details?id=com.sylkat.apache ). Now, it's not really as comfortable as Eclipse for me, but I'll live with that.
> 
> Long story: Eclipse requires a Web Browser to be able to launch a project you're working to from its IDE, and the Browser itself you can install only with Snap (I tried `sudo apt install firefox` or `sudo apt install chromium-browser`, but both it only downloads a dummy package wich asks to download the Snap for the equivalent). Hence, the requirement for Snap.
> 
> Meanwhile, VHEditor is a port of CodeServer to Android (it's rather more of an adaptation with a WebView), and it works as a native Android App, so I can access my internal memory without weird mechanisms like termux-setup-storage. The AWebServer App provides PHP, MySQL and Apache, enabling me to select my working folder case-by-case (based on my project folder) and allows me to code simple, yet complete Web Apps with PHP/MySQL.
> 
> Thanks anyhow, and I appreciate the time you took for helping me.

I understand, but this is unfortunately a very Ubuntu specific issue that is also referenced on a lot of forums and sites all over the web.

It is still strongly suggested to use the tools made available in the repositories as they will always integrate best with the deployment: external packages are to be used at your own risk and with prior knowledge of what you need to do to get them working. Another alternative would be to get them to compile for either Termux or arm64 ubuntu if they aren't available.

I've added a removal instruction for snap during setup and will be updating the readme to reflect this.

Thanks for your input!

**Reply by anonimo82** [reply]
Man, your Dextop is a welcome novelty in Termux, way better than proot-distro and Andronix.
I guess this is THE way to go for a full blown distro, it's a great start.
IMHO, just add virtual devices by mapping Termux:API functions to /dev/whatever and support 3D hardware acceleration with Tewmux disabled repo (see my posts here: https://github.com/nathaneltitane/dextop/discussions/8 )

Thanks a lot for making this possible in the first place. =)

**Reply by nathaneltitane** [reply]
> Man, your Dextop is a welcome novelty in Termux, way better than proot-distro and Andronix. I guess this is THE way to go for a full blown distro, it's a great start. IMHO, just add virtual devices by mapping Termux:API functions to /dev/whatever and support 3D hardware acceleration with Tewmux disabled repo (see my posts here: #8 )
> 
> Thanks a lot for making this possible in the first place. =)

I appreciate your words and input very much!

I haven't had much time to delv into those but I can assure your I will be taking care of it as Dextop is currently very usable but far from what I intend it to be.

I made this project because of how cumbersome and unintuitive other deployments and attempts were at iterating the dextop experience onto Android devices.

More to come soon, hopefully!

**Reply by anonimo82** [reply]
Yeah, hopefully. Me hoping too =)
Keep up this great work
